extensions/email/src/main/java/org/mule/extension/email/internal/util/EmailConnectorUtils.java

```diff
@@ -0,0 +1,223 @@
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.email.internal.util;
+
+import static java.lang.String.format;
+import static javax.mail.Part.ATTACHMENT;
+import static org.mule.runtime.core.util.IOUtils.toByteArray;
+import org.mule.extension.email.api.EmailAttributes;
+import org.mule.extension.email.internal.exception.EmailException;
+import org.mule.runtime.api.message.MuleMessage;
+
+import java.io.ByteArrayInputStream;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+
+import javax.activation.DataHandler;
+import javax.mail.Address;
+import javax.mail.BodyPart;
+import javax.mail.MessagingException;
+import javax.mail.Multipart;
+import javax.mail.Part;
+import javax.mail.internet.AddressException;
+import javax.mail.internet.InternetAddress;
+
+/**
+ * this class contains common methods for email handling.
+ *
+ * @since 4.0
+ */
+public final class EmailConnectorUtils
+{
+    public static final String EIGHT_BIT_ENCODING = "ISO-8859-1";
+    public static final String HEADERS_SEPARATOR = "\r\n\r\n";
+    public static final String ATTRIBUTES_NOT_FOUND_MASK = "No email attributes were found in the incoming message. %s";
+
+    /**
+     * Hide constructor
+     */
+    private EmailConnectorUtils()
+    {
+    }
+
+    /**
+     * Converts a simple {@link String} representing an address into an {@link InternetAddress} instance
+     *
+     * @param address the string to be converted.
+     * @return a new {@link InternetAddress} instance.
+     */
+    public static Address toAddress(String address)
+    {
+        try
+        {
+            return new InternetAddress(address);
+        }
+        catch (AddressException e)
+        {
+            throw new EmailException(format("Error while creating %s InternetAddress", address));
+        }
+    }
+
+    /**
+     * Converts a {@link List} of {@link String}s representing email addresses into an {@link InternetAddress} array.
+     *
+     * @param addresses the list to be converted.
+     * @return a new {@link Address}[] instance.
+     */
+    public static Address[] toAddressArray(List<String> addresses)
+    {
+        return addresses.stream().map(EmailConnectorUtils::toAddress).toArray(Address[]::new);
+    }
+
+    /**
+     * Extracts the incoming {@link MuleMessage} attributes of {@link EmailAttributes} type.
+     * <p>
+     * If no {@link EmailAttributes} are found in the incoming {@link MuleMessage}
+     * an exception with the {@code exceptionMessage} is thrown.
+     *
+     * @param muleMessage      the incoming {@link MuleMessage}.
+     * @param exceptionMessage i
+     * @return an value with the {@link EmailAttributes}.
+     */
+    public static EmailAttributes getAttributesFromMessage(MuleMessage muleMessage, String exceptionMessage)
```

Comment:
I don't think that this should throw exception. I would rather have it return an `Optional`